hail/src/main/scala/is/hail/annotations/aggregators/RegionValueApproxCDFAggregator.scala

```diff
@@ -248,18 +252,15 @@ class ApproxCDFCombiner[@specialized(Int, Long, Float, Double) T: ClassTag : Ord
   /* Compact level `level`, merging the compacted results into level `level+1`.
    * Shift lower levels up to keep items contiguous.
    */
-  def compactLevel(level: Int) {
-    compactLevel(level, true)
-  }
-
-  def compactLevel(level: Int, shiftLowerLevels: Boolean) {
+  def compactLevel(level: Int, shiftLowerLevels: Boolean = true, dummy: T = helper.dummyValue) {
```

Comment:
What's the purpose of these dummy variables? I don't see anywhere these functions are overloaded?

Reply:
Without the dummy variable, this function would have no `T` in its signature, which means Scala would refuse to generate specialized methods. But we very much want the implementation of `_compactLevel` to be specialized for each concrete type, to avoid boxing, and unspecialized methods always call unspecialized versions of any called methods. There were just a few spots I missed adding dummy parameters in the original PR.

Reply:
ah, cool!